Make airship control plane node count configurable

jesusaurus opened the pull request:
Add an ansible variable for the number of caasp workers to use as
airship control plane workers instead of hard-coding to 2.

jsuchome commented:
1. Could you also adapt playbooks/roles/caasp4/templates/inventory-caasp4.yml.j2 ?
2. What if we are not deploying on openstack?

jesusaurus commented:
@jsuchome
1. done
2. I'm not very familiar with that use case, I thought inventory files were manually created in that case. I've renamed the variable and moved it to common-vars

jsuchome commented on vars/common-vars.yml:
```diff
@@ -125,3 +125,5 @@ scale_profile: "{{ lookup('env','scale_profile') | default('minimal', true) }}"
 #Set timeout for helm cleanup.
 helm_delete_timeout: 900
+
+airship_ucp_workers: 2
```
The meaning is actually "control workers", right? As opposed to "compute workers", if I get it right. So it is a subset of "caasp workers" and current variable name does not make it clear IMHO

jsuchome commented on playbooks/openstack-caasp_instance.yml:
```diff
@@ -116,8 +116,8 @@
               airship-openstack-control-workers:
-                hosts: &firsttwoworkers
-              {% for node in workers %}{% if loop.index <= 2 %}
+                hosts: &first_workers
```
What is actually the meaning of this? I do not understand the syntax...

jesusaurus replied:
This is a yaml "anchor" 
https://confluence.atlassian.com/bitbucket/yaml-anchors-960154027.html

jsuchome replied:
Thanks!

toabctl commented on vars/common-vars.yml:
```diff
@@ -143,6 +143,10 @@ scale_profile: "{{ lookup('env','scale_profile') | default('minimal', true) }}"
 
+# Number of caasp workers to use for the control plane,
+# the remainder will be used as computer hosts
+airship_ucp_control_workers: 2
```
Why common-vars? This is airship specific...

jesusaurus replied:
I put it under the airship-only section of common-vars https://github.com/SUSE-Cloud/socok8s/blob/master/vars/common-vars.yml#L56; we can move that entire section to a new file, but I believe that is a separate task from this PR.

toabctl commented on playbooks/roles/caasp4/templates/inventory-caasp4.yml.j2:
```diff
@@ -32,15 +32,15 @@ airship-openstack-control-workers:
   hosts:
 {% for worker_ip in _terraform_json_output.ip_workers.value %}
-{% if loop.index0 > 1 %}
+{% if loop.index0 >= airship_ucp_control_workers %}
```
urgs. so we (again) include $random variables from everywhere in specific roles. But not your fault..
